microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/java/hover/PropertiesHoverParticipant.java

```diff
@@ -74,8 +76,21 @@ public PropertiesHoverParticipant(String annotationName, String annotationMember
 	public PropertiesHoverParticipant(String annotationName, String annotationMemberName,
 			String defaultValueAnnotationMemberName) {
 		this.annotationName = annotationName;
-		this.annotationMemberName = annotationMemberName;
 		this.defaultValueAnnotationMemberName = defaultValueAnnotationMemberName;
+		this.annotationMembers = new String [] { annotationMemberName };
+	}
+
+	/**
+	 * The definition participant constructor.
+	 *
+	 * @param annotationName                   the annotation name (ex :
+	 *                                         io.quarkus.scheduler.Scheduled)
+	 * @param annotationMembers                a list of annoation members (ex :

```

Comment:
annotation

Reply:
@rgrunber what about having a main constructor like:

```java
PropertiesHoverParticipant(String annotationName, String [] annotationMemberNames, String [] annotationMemberValues)
```

and after you provide another constructor which call this main constructor?

Reply:
Done. the initialization is done by a single constructor.